pkg/operator/test/client.go

```diff
@@ -46,7 +46,9 @@ func New(nodes int) *fake.Clientset {
 				},
 			},
 		}
-		clientset.CoreV1().Nodes().Create(n)
+		if _, err := clientset.CoreV1().Nodes().Create(n); err != nil {
+			return nil, fmt.Errorf("failed to create fake K8s cluster. %v", err)
```

Comment:
Instead of changing this method to return error, this is also a good place to pass in the testing context so we can just immediately assert if there is a failure. No need for all the callers to assert if we can assert in this one place.